crates/ark/src/analysis/parse_boundaries.rs

```diff
@@ -50,19 +76,20 @@ pub fn parse_boundaries(text: &str) -> anyhow::Result<ParseBoundaries> {
 
     let mut complete: Vec<LineRange> = vec![];
     let mut incomplete: Option<LineRange> = None;
-    let mut error: Option<LineRange> = None;
+    let mut invalid: Option<LineRange> = None;
+    let mut invalid_message: Option<String> = None;
```

Comment:
Main comment is maybe you could make this an optional tuple like `Option<(LineRange, String)>` so you don't have to `unwrap()` the `invalid_message` later on (which made me stop and think for a second)